src/app/shared/directives/search-field-validation.directive.ts

```diff
@@ -0,0 +1,23 @@
+import { Directive, ElementRef, EventEmitter, HostListener, Output } from '@angular/core';
+
+@Directive({
+  selector: '[appSearchFieldValidation]'
+})
+export class SearchFieldValidationDirective {
```

Comment:
_Left a comment here to keep everything in one thread_

My main point why this is not the best solution for us is that it is better to use common practices for reactive forms. In our application, we usually use validators to check if the value is correct, which is a straightforward, so I don't think we should change it to any other approach.